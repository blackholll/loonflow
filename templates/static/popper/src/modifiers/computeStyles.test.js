// @flow
import { mapToStyles } from './computeStyles';

it('computes the popper styles', () => {
  window.devicePixelRatio = 2;

  expect(
    mapToStyles({
      popper: document.createElement('div'),
      placement: 'bottom',
      popperRect: { x: 10, y: 10, width: 100, height: 100 },
      offsets: { x: 10, y: 5 },
      position: 'absolute',
      gpuAcceleration: true,
      adaptive: true,
      roundOffsets: true,
    })
  ).toMatchSnapshot();

  expect(
    mapToStyles({
      popper: document.createElement('div'),
      placement: 'bottom',
      popperRect: { x: 10, y: 10, width: 100, height: 100 },
      offsets: { x: 10, y: 5 },
      position: 'absolute',
      gpuAcceleration: false,
      adaptive: true,
      roundOffsets: true,
    })
  ).toMatchSnapshot();

  expect(
    mapToStyles({
      popper: document.createElement('div'),
      placement: 'top',
      popperRect: { x: 10, y: 10, width: 100, height: 100 },
      offsets: { x: 10, y: 5 },
      position: 'absolute',
      gpuAcceleration: false,
      adaptive: true,
      roundOffsets: true,
    })
  ).toMatchSnapshot();

  expect(
    mapToStyles({
      popper: document.createElement('div'),
      placement: 'left',
      popperRect: { x: 10, y: 10, width: 100, height: 100 },
      offsets: { x: 10, y: 5 },
      position: 'absolute',
      gpuAcceleration: false,
      adaptive: true,
      roundOffsets: true,
    })
  ).toMatchSnapshot();

  window.devicePixelRatio = 1;
});

it('computes the arrow styles', () => {
  expect(
    mapToStyles({
      popper: document.createElement('div'),
      placement: 'bottom',
      popperRect: { x: 10, y: 10, width: 100, height: 100 },
      offsets: { x: 10, y: 5 },
      position: 'absolute',
      gpuAcceleration: true,
      adaptive: false,
      roundOffsets: true,
    })
  ).toMatchSnapshot();
});
