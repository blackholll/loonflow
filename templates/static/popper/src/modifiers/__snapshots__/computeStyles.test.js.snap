// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computes the arrow styles 1`] = `
Object {
  "left": "0",
  "position": "absolute",
  "top": "0",
  "transform": "translate(10px, 5px)",
}
`;

exports[`computes the popper styles 1`] = `
Object {
  "bottom": "auto",
  "left": "0",
  "position": "absolute",
  "right": "auto",
  "top": "0",
  "transform": "translate3d(10px, 5px, 0)",
}
`;

exports[`computes the popper styles 2`] = `
Object {
  "bottom": "auto",
  "left": "10px",
  "position": "absolute",
  "right": "auto",
  "top": "5px",
  "transform": "",
}
`;

exports[`computes the popper styles 3`] = `
Object {
  "bottom": "-105px",
  "left": "10px",
  "position": "absolute",
  "right": "auto",
  "top": "auto",
  "transform": "",
}
`;

exports[`computes the popper styles 4`] = `
Object {
  "bottom": "auto",
  "left": "auto",
  "position": "absolute",
  "right": "-110px",
  "top": "5px",
  "transform": "",
}
`;
